packages/experimental/src/user-interaction-instrumentation.ts

```diff
@@ -0,0 +1,206 @@
+import { context, Span, trace } from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+import { getElementXPath } from '@opentelemetry/sdk-trace-web';
+
+import { VERSION } from './version';
+
+const INSTRUMENTATION_NAME = '@honeycombio/user-instrumentation';
+const SPAN_KEY = '__HNY_SPAN';
+
+const DEFAULT_EVENT_NAMES: EventName[] = ['click'];
+
+type EventName = keyof GlobalEventHandlersEventMap;
+
+type Listener = {
+  eventName: EventName;
+  handler: (event: Event) => void;
+};
+
+interface UserInteractionInstrumentationConfig extends InstrumentationConfig {
+  eventNames?: EventName[];
+  rootNodeId?: string;
+}
+
+export class UserInteractionInstrumentation extends InstrumentationBase {
+  protected _config: UserInteractionInstrumentationConfig;
+  private _is_enabled: boolean;
+  private _listeners: Listener[];
+
+  constructor(config: UserInteractionInstrumentationConfig = {}) {
+    super(INSTRUMENTATION_NAME, VERSION, config);
+    this._config = config;
+    this._is_enabled = this._config.enabled ?? false;
```

Comment:
🤔 usually we'd do `_isEnabled` but here we're doing 🐍 casing. Is that because of a collision with `InstrumentationBase`? I think I ran into a similar issue with `GlobalErrorInstrumentation`. I wasn't getting much value out off the base class and it wasn't reporting enabled correctly, so I just implemented `InstrumentationAbstract` instead.

Reply:
nah this might've just me been losing track, will update